project/dexter-vd-cpp/src/java/com/samsung/sec/dexter/vdcpp/util/CDTASTUtil.java

```diff
@@ -0,0 +1,401 @@
+
+	public static boolean isLongLong(IASTNode node){
+		IASTName name=null;
+		if(node instanceof IASTIdExpression){	
+			name =((IASTIdExpression) node).getName();
+		}else if(node instanceof IASTDeclarator){
+			name =((IASTDeclarator) node).getName();
+		}
+		
+		if(name==null)	return false; 								
+		final IBinding binding = name.resolveBinding();
+		if (binding == null)	
+			return false;
+		
+		if(!(binding instanceof IVariable))
+			return false;
+		
+		IType type=((IVariable) binding).getType();
+		return (type instanceof IBasicType && ((IBasicType) type).isLongLong());
+	}
+	
+	public static boolean isLong(IASTNode node){
+		IASTName name=null;
+		if(node instanceof IASTIdExpression){	
+			name =((IASTIdExpression) node).getName();
+		}else if(node instanceof IASTDeclarator){
+			name =((IASTDeclarator) node).getName();
+		}
+		
+		if(name==null)	return false; 								
+		final IBinding binding = name.resolveBinding();
+		if (binding == null)	
+			return false;
+		
+		if(!(binding instanceof IVariable))
+			return false;
+		
+		IType type=((IVariable) binding).getType();
+		return (type instanceof IBasicType && ((IBasicType) type).isLong());		
+	}
+	
+	public static boolean isSigned(IASTNode node){
+		IASTName name=null;
+		if(node instanceof IASTIdExpression){	
+			name =((IASTIdExpression) node).getName();
+		}else if(node instanceof IASTDeclarator){
+			name =((IASTDeclarator) node).getName();
+		}
+		
+		if(name==null)	return false; 								
+		final IBinding binding = name.resolveBinding();
+		if (binding == null)	
+			return false;
+		
+		if(!(binding instanceof IVariable))
+			return false;
+		
+
+		IType type=((IVariable) binding).getType();
+		
+		return !ASTTypeUtil.getType(type).contains("unsigned");	
+	}
+	
+	public static boolean isUnsigned(IASTNode node){
+		IASTName name=null;
+		if(node instanceof IASTIdExpression){	
+			name =((IASTIdExpression) node).getName();
+		}else if(node instanceof IASTDeclarator){
+			name =((IASTDeclarator) node).getName();
+		}
+		
+		if(name==null)	return false; 								
+		final IBinding binding = name.resolveBinding();
+		if (binding == null)	
+			return false;
+		
+		if(!(binding instanceof IVariable))
+			return false;
+		
+		IType type=((IVariable) binding).getType();
+		
+		return ASTTypeUtil.getType(type).contains("unsigned");	
+	}
+	
+	public static boolean checkVariableType(IASTNode node, String regex){
+		IASTName name=null;
+		if(node instanceof IASTIdExpression){	
+			name =((IASTIdExpression) node).getName();
+		}else if(node instanceof IASTDeclarator){
+			name =((IASTDeclarator) node).getName();
+		}
+		
+		if(name==null)	return false; 								
+		final IBinding binding = name.resolveBinding();
+		if (binding == null)	
+			return false;
+		
+		if(!(binding instanceof IVariable))
+			return false;
+		
+		IType type=((IVariable) binding).getType();				
+		return getBasicType(type).matches(regex);
+	}
+	
+	public static String getBasicType(IType type){
+		String typeString="";
```

Comment:
you can use enum instead of returning String.
for example.
static enum PrimitiveType = { CHAR, SHORT, INT };